samcli/commands/validate/validate.py

```diff
@@ -49,13 +51,20 @@ def do_cli(ctx, template):
     sam_template = _read_sam_file(template)
 
     iam_client = boto3.client("iam")
-    validator = SamTemplateValidator(sam_template, ManagedPolicyLoader(iam_client))
+    validator = SamTemplateValidator(
+        sam_template, ManagedPolicyLoader(iam_client), profile=ctx.profile, region=ctx.region
+    )
 
     try:
         validator.is_valid()
     except InvalidSamDocumentException as e:
         click.secho("Template provided at '{}' was invalid SAM Template.".format(template), bg="red")
         raise InvalidSamTemplateException(str(e)) from e
+    except NoRegionFound as no_region_found_e:
+        raise UserException(
+            "AWS Region was not found. Please configure your region through a profile or --region option",
```

Comment:
should we add AWS_REGION env var too?

Reply:
I am not sold on adding [`AWS_DEFAULT_REGION`](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#guide-configuration). Yes customers can set this as it is resolved by boto3, but I don't see the need to add every single option here. Most customers are going to use `--region` or define them through a profile locally.

Is there a specific reason why you think we should add it?

Reply:
does sam validate automatically pick up the env vars? just trying to see if we can reduce friction where we can.

Reply:
Yes. We create a `boto_session` like we do in other places throughout. When `None` is passed, boto will resolve for us.